### PR TITLE
Set Go support policy to latest-2

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Before updating deps, we should revisit the go support policy I think. That way the next release is coherent across the board. This is the same templating as wazero, which had to deal with some projects that lagged behind Go.

### CODE REVIEW

**Review by basvanbeek — COMMENTED** [review]
This CI flow removes Windows and Mac OS testing. Was this broken?

**Comment by codefromthecrypt** [comment]
> This CI flow removes Windows and Mac OS testing. Was this broken?

nope that was a mistake. I'll fix that, sorry for the distraction!

**Comment by codefromthecrypt** [comment]
I'll raise a separate PR first for lint as seems we can't pass it anymore

File: .github/workflows/ci.yml
Comment (codefromthecrypt) [inline .github/workflows/ci.yml:16]: ps after rebasing on lint, I'll put back to the other style, just change version to matrix (leave the other things alone notably that we are only testing old versions with ubuntu)

**Comment by codefromthecrypt** [comment]
ok I removed the fuzz so this does only one thing: formalizes the current-2 version policy from the github action config.

This should pass once #221 is merged

**Comment by codefromthecrypt** [comment]
we need to bump the floor at least to Go 1.19 in order to fix the CVEs. Would like an approve here as this seems a no brainer, and if not, we should say why https://github.com/openzipkin/zipkin-go/pull/222

**Comment by codefromthecrypt** [comment]
TL;DR; let's solidify status quo of our tests to latest-2 or document why not. latest-* isn't a policy, we should have a policy as most maintainers are dormant. This allows people who do maintain things frequently like @anuraaga, @reta and I proceed without others and without judgemental mistakes.

If I see no feedback to contrary in a few days, I'll merge as I can't block on things like version bumps as they don't deserve to tie any of our time up in nagging. We need to block our volunteers the least as possible on mechanical work.

**Review by anuraaga — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
thanks for the look @anuraaga @basvanbeek and @reta! I'll rebase the other PR so that we can kill that CVE. If someone later complains it should be latest-3 we can revisit.

**Pull request merged**
